docs/deploying.md

```diff
@@ -128,95 +128,85 @@ This snippet assumes the following resources are already defined:
-4. Add the default route to your route table on the subnet. It is best to do this manually so you can do a seamless cut over from your existing nat gateway. Go to `VPC > Route Tables > Private route table > Routes > Edit Routes` Add a 0.0.0.0/0 route pointing to the network interface.
+3. Deploy with CloudFormation `aws cloudformation deploy --force-upload --capabilities CAPABILITY_IAM --template-file template.yml --stack-name FckNat`
+4. Add the default route to your route table on the subnet. It is best to do this manually so you can do a seamless cut over from your existing NAT gateway. Go to `VPC > Route Tables > Private route table > Routes > Edit Routes` Add a 0.0.0.0/0 route pointing to the network interface.
 
 ``` yaml
 Parameters:
-  vpc:
-    Type: String
-    Default: "vpc-121212121212121212"
-  subnet:
-    Type: String
-    Default: "subnet-121212121212121212"
-  CIDR:
+  VpcIdParameter:
+    Type: AWS::EC2::VPC::Id
+  SubnetIdParameter:
+    Type: AWS::EC2::Subnet::Id
+  CIDRParameter:
     Type: String
     Default: "10.0.0.0/16"
+  FckNatAMIParameter:
+    Type: AWS::EC2::Image::Id
+    Default: ami-075a0093cd9926d44
 
 Resources:
   FckNatInterface:
     Type: AWS::EC2::NetworkInterface
     Properties:
-      SubnetId: !Sub "${subnet}"
+      Description: FckNat Gateway Interface
+      SubnetId: !Ref SubnetIdParameter
       GroupSet:
-        - Fn::GetAtt:
-            - NatSecurityGroup
-            - GroupId
+        - !GetAtt [FckNatSecurityGroup, GroupId]
       SourceDestCheck: false
-      
   FckNatAsgInstanceProfile:
     Type: AWS::IAM::InstanceProfile
     Properties:
       Roles:
-        - Ref: NatRole
-
+        - !Ref FckNatRole
   FckNatLaunchTemplate:
     Type: AWS::EC2::LaunchTemplate
+    DependsOn: FckNatRole
     Properties:
       LaunchTemplateName: FckNatLaunchTemplate
       LaunchTemplateData:
-        ImageId: ami-05b6d5a2e26f13c93
+        ImageId: !Ref FckNatAMIParameter
         InstanceType: t4g.nano
         IamInstanceProfile:
-          Ref: FckNatAsgInstanceProfile
-        SecurityGroups:
-          - Fn::GetAtt:
-              - NatSecurityGroup
-              - GroupId
+          Name: !Ref FckNatAsgInstanceProfile
+        SecurityGroupIds:
+          - !GetAtt [FckNatSecurityGroup, GroupId]
         UserData:
-          Fn::Base64:
-            Fn::Join:
-              - ""
-              - - |-
-                  #!/bin/bash
-                  echo "eni_id=
-                - Ref: FckNatInterface
-                - |-
-                  " >> /etc/fck-nat.conf
-                  service fck-nat restart
-    DependsOn:
-      - NatRole
-
+          Fn::Base64: !Sub |
+            #!/bin/bash
+            echo "eni_id=${FckNatInterface}" >> /etc/fck-nat.conf
+            service fck-nat restart
   FckNatAsg:
     Type: AWS::AutoScaling::AutoScalingGroup
     Properties:
       MaxSize: "1"
       MinSize: "1"
-      DesiredCapacity: "1"
```

Comment:
I believe this is necessary?

Reply:
I guess I should say _necessary in fck-nat's case_ since we always and forever only want to have one instance at a time. If I remember right there's some weirdness with `MinSize` and `MaxSize`, but it's been a while since I've had to interface with them directly.

Reply:
I don't believe it's necessary, but it's also harmless when everything is `1`. Happy to put it back if you'd prefer.

Reply:
Let's do that just to be safe